### PR TITLE
Migrates spot-extract to use Clize.

### PR DESCRIPTION
**Pull request by MoseleyS**
Testing:
 - [x] Ran tests and they passed OK

### CODE REVIEW

File: lib/improver/cli/spot_extract.py [hunk 1]
```diff
@@ -31,145 +31,21 @@
...
-def process(neighbour_cube, diagnostic_cube, lapse_rate_cube=None,
-            apply_lapse_rate_correction=False, land_constraint=False,
-            minimum_dz=False, extract_percentiles=None,
-            ecc_bounds_warning=False, new_title=None,
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(neighbour_cube: cli.inputcube,
```
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many local variables (30/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 5
           

Complexity increasing per file
==============================
- lib/improver/cli/spot_extract.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289&bid=15658332)

File: lib/improver/cli/spot_extract.py [hunk 2]
```diff
@@ -223,15 +100,17 @@ def process(neighbour_cube, diagnostic_cube, lapse_rate_cube=None,
             Default is False.
         new_title (str or None):
             New title for the spot-extracted data.  If None, this attribute is
-            removed from the output cube.
+            removed from the output cube since it has no prescribed standard 
+            and may therefore contain grid information that is no longer
```
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

File: lib/improver/cli/spot_extract.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many statements (59/50)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many branches (17/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

File: lib/improver/cli/spot_extract.py [hunk 3]
```diff
@@ -223,15 +100,17 @@ def process(neighbour_cube, diagnostic_cube, lapse_rate_cube=None,
             Default is False.
         new_title (str or None):
             New title for the spot-extracted data.  If None, this attribute is
-            removed from the output cube.
+            removed from the output cube since it has no prescribed standard
```
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity increasing per file
==============================
- lib/improver/cli/spot_extract.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289&bid=15658332)

File: lib/improver/cli/spot_extract.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many branches (17/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many local variables (30/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many statements (59/50)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity increasing per file
==============================
- lib/improver/cli/spot_extract.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289&bid=15658332)

File: lib/improver/cli/spot_extract.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many statements (59/50)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many local variables (30/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many branches (17/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

File: lib/improver/cli/spot_extract.py [hunk 4]
```diff
@@ -31,145 +31,21 @@
...
-                     args.apply_lapse_rate_correction, args.land_constraint,
-                     args.minimum_dz, args.extract_percentiles,
-                     args.ecc_bounds_warning, args.new_title,
-                     args.suppress_warnings)
-
-    # Save Cube
-    save_netcdf(result, args.output_filepath)
-
-
-def process(neighbour_cube, diagnostic_cube, lapse_rate_cube=None,
```
Comment (fionaRust) [inline lib/improver/cli/spot_extract.py]: I think the docstring for the lapse rate cube should say it is optional

File: lib/improver/tests/acceptance/test_spot_extract.py [hunk 5]
```diff
@@ -206,8 +206,8 @@ def test_percentile_deterministic(tmp_path):
     diag_path = kgo_dir / "inputs/ukvx_temperature.nc"
     kgo_path = kgo_dir / "outputs/nearest_uk_temperatures.nc"
     output_path = tmp_path / "output.nc"
-    args = [neighbour_path, diag_path, output_path,
-            "--extract_percentiles", "50"]
+    args = [neighbour_path, diag_path, "--output", output_path,
+            "--extract-percentiles", "50"]
     with pytest.warns(None):
```
Comment (fionaRust) [inline lib/improver/tests/acceptance/test_spot_extract.py]: When looking through these tests quickly I noticed this one doesn't check the text of the warning and the rest of them do. Should we fix this?

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity increasing per file
==============================
- lib/improver/cli/spot_extract.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289&bid=15658332)

File: lib/improver/cli/spot_extract.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many statements (59/50)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many local variables (30/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many branches (17/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 3
           

Complexity increasing per file
==============================
- lib/improver/tests/acceptance/test_spot_extract.py  1
- lib/improver/cli/spot_extract.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289&bid=15658332)

File: lib/improver/cli/spot_extract.py (hunk 1 above)
Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many statements (59/50)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many local variables (30/20)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

Comment (benfitzpatrick) [inline lib/improver/cli/spot_extract.py]: Codacy Issue found: [Too many branches (17/12)](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4681289)

**Comment by MoseleyS** [comment]
Updated docstrings as requested.

**Comment by fionaRust** [comment]
@MoseleyS travis isn't failing because of your change. I've agree with @gavinevans that I'll do a quick PR to fix the feature branch

**Pull request merged**